src/ftest.jl

```diff
@@ -0,0 +1,155 @@
+                     Result=[1.1, 1.2, 1, 2.2, 1.9, 2, .9, 1, 1, 2.2, 2, 2]);
+julia> mod = lm(@formula(Result~Treatment), dat        );
+julia> nullmod = lm(@formula(Result~1), d);
+julia> ft = ftest(mod.model, nullmod.model)
+         Residual DOF DOF ΔDOF      SSR    ΔSSR           R²       ΔR²      F*      p(>F)
+Model 1          10.0 3.0  NaN 0.128333     NaN     0.960258       NaN     NaN        NaN
+Model 2          11.0 2.0 -1.0  3.22917 3.10083 -2.22045e-16 -0.960258 241.623 2.48122e-8
+
+
+```
+"""
+function ftest(mods::LinearModel...)
+    nmodels = length(mods)
+    for i in 2:nmodels
+        issubmodel(mods[i], mods[i-1]) || 
+        throw(ArgumentError("F test $i is only valid if model $i is nested in model $i-1"))
+    end
+
+    SSR = deviance.(mods)
+
+    nparams = dof.(mods)
+
+    df2 = _diffn(nparams)
+    df1 = Int.(dof_residual.(mods))
+
+    MSR1 = _diff(SSR)./df2
+    MSR2 = (SSR./df1)[1:nmodels-1]
+
+    fstat = MSR1./MSR2
+    pval = PValue.(ccdf.(FDist.(df2, df1[1:nmodels-1]), fstat))
+    return FTestResult(SSR, nparams, df1, r2.(mods), fstat, pval)
+#=
+    prepend!(pval, [NaN])
+    prepend!(fstat, [NaN])
+
+    df2 = Float64.(df2)
+    prepend!(df2, [NaN])
+
+    R2 = collect(r2.(mods))
+    ΔR2 = -diff(R2)
+    prepend!(ΔR2, [NaN])
+
+    ΔSSR = -diff(SSR)
+    prepend!(ΔSSR, [NaN])
+    return CoefTable([df1, nparams, df2, SSR, ΔSSR, R2, ΔR2, fstat, pval],
+                     ["Res. DOF", "DOF", "ΔDOF", "SSR", "ΔSSR", "R²", "ΔR²", "F*",
+                     "p(>F)"],
+                     ["Model $i" for i in 1:nmodels])=#
+end
+
+function show{N}(io::IO, ftr::FTestResult{N})
+    Δdof = _diffn(ftr.dof_resid)
+    Δssr = _diffn(ftr.ssr)
+    ΔR² = _diffn(ftr.r2)
+
+    nc = 10
+    nr = N
+    outrows = Array{String, 2}(nr+1, nc)
+    
+    outrows[1, :] = ["", "Res. DOF",  "DOF",  "ΔDOF",  "SSR",
+                    "ΔSSR",  "R²",  "ΔR²",  "F*",  "p(>F)"]
+    outrows[2, :] = ["Model 1", @sprintf("%.4f", ftr.dof_resid[1]),
+                     @sprintf("%.4f", ftr.nparams[1]), " ", @sprintf("%.4f", ftr.ssr[1]),
+                     " ", @sprintf("%.4f", ftr.r2[1]), " ", " ", " "]
+    
+    for i in 2:nr
+        outrows[i+1, :] = ["Model $i", @sprintf("%.4f", ftr.dof_resid[i]),
+                           @sprintf("%.4f", ftr.nparams[i]), @sprintf("%.4f",
+                           Δdof[i-1]), @sprintf("%.4f", ftr.ssr[i]),
+                           @sprintf("%.4f", Δssr[i-1]), @sprintf("%.4f", ftr.r2[i]),
+                           @sprintf("%.4f", ΔR²[i-1]), @sprintf("%.4f", ftr.fstat[i-1]),
+                           string(ftr.pval[i-1])]
+    end
+    colwidths = length.(outrows)
+    max_colwidths = [maximum(view(colwidths, :, i)) for i in 1:nc]
+
+    for r in 1:nr+1
+        for c in 1:nc
+            cur_cell = outrows[r, c]
+            cur_cell_len = length(cur_cell)
+            
+            print(io, cur_cell)
+            print(io, RepString(" ", max_colwidths[c]-cur_cell_len+1))
```

Comment:
Use `" "^(max_colwidths[c]-cur_cell_len+1)` as `RepString` is likely to be deprecated at some point.